coordinate-systems.md

```diff
@@ -0,0 +1,246 @@
+
+Option (4) does exactly that - removes XRFrame and just assumes we are always talking about the current frame. All APIs are simplified, for example:
+
+
+```
+xrSession.setDevicePoseMode("stage");
+const devicePose = xrSession.getDevicePose();
+```
+
+
+Options (2), (3), and (4) all appear to be valid. Option (2) and (3) mainly differ in whether old XRFrame objects are invalidated or not. Option (3) removes some of the "don't do that" edge-case concerns, however it re-introduces the problem that XRFrame is meant to solve - that different platforms might not be able to handle calls dealing with world state between frames gracefully. Option (4) seems strictly better than (3) in that it at least simplifies the API surface greatly. So the question is really between (2) and (4) - 
+
+
+## Relative Hit-test API
+
+The problem of how to do pixel-perfect reticle functionality is a major use case of these APIs. The problem statement can be found [here](https://docs.google.com/document/d/1cGLkmihH4yZy5UlwyBSszWk0GGb8JJ4OGGI-INBX3YM/edit).
+
+In order to do a hit-test with a relative ray, we did an exploration of API shape given the current definitions/relationships of XRFrame, XRFOR, and XRCoordinateSystem. This is what we came up with:
+
+
+```
+XRSession.requestHitTest(XRRay ray, XRCoordinateSystem cs, (bool success, XRFrame frame, Array<XRHitResult> results) => {});
+XRHitResult.getMatrix(XRFrame, XRCoordinateSystem);
+```
+
+
+Let's go through each parameter and how it is used:
+
+*   input
+    *   ray: the ray to cast into the world and collect hits
+    *   coordinate system: represents the coordinate system to which the ray is relative. This is needed in order to be able to pass in a ray that is device-relative and get an accurate result for the future frame in which the calculation was made.
+*   callback
+    *   success: a boolean saying whether there was an error in the request. This is needed since we can no longer use a promise if we return an XRFrame, as XRFrames are not valid except within the callback where they are issued and thus, do to the ability to store a promise for later use, would be invalid in a late-bound promise.
+    *   frame: the frame for which the results are valid. This is needed as the current API shape requires XRFrame objects to resolve transforms of hits and the device
```

Comment:
The `XRHitResult` already represents a precise moment in time for which the hit-test was performed: the forward-predicted device position for the frame after the request was made.  It is not meaningful for the app to supply the `XRFrame` again to `XRHitResult.getMatrix`, as it's not clear what the app would mean if it specified a different time for an already-resolved hit result.

From the discussion above, it sounds like one of the fundamental WebXR issues we need to nail down regards the lifetime of `XRFrame` objects, and what processed data they keep alive when apps hold onto them, as there are many follow-on impacts if we define WebXR to only allow one frame of resolved pose data to be alive at a time.